lib/msf/base/sessions/command_shell.rb

```diff
@@ -193,12 +196,28 @@ def shell_write(buf)
   # :category: Msf::Session::Provider::SingleCommandShell implementors
   #
   # Closes the shell.
+  # Note: parent's 'self.kill' method calls cleanup below.
```

Comment:
This is important to note, since the parent's kill command will call cleanup, which means we didn't need to continue doing a direct rstream.close

Reply:
This is a nauseatingly common problem with our setups and cleanups. Good catch.